urnai/sc2/states/terran_state.py

```diff
@@ -0,0 +1,121 @@
+import math
+
+import numpy as np
+from pysc2.env import sc2_env
+from pysc2.lib import features, units
+
+from urnai.states.state_base import StateBase
+
+
+class TerranState(StateBase):
+    
+    def __init__(self, grid_size=4):
+
+        self.grid_size = grid_size
+        # Size of the state returned
+        # 22: number of data added to the state (number of minerals, army_count, etc)
+        # 2 * 4 ** 2: 2 grids of size 4 x 4, representing enemy and player units
+        self._state_size = int(22 + 2 * (self.grid_size ** 2))
+        self.player_race = sc2_env.Race.terran
+        self.base_top_left = None
```

Comment:
Why `base_top_left`?

Reply:
I understood that this variable is a boolean that indicates if the townhall is located at the top left of the map. That would mean having an x and y with at least half of the size of the map, even though it only considers the x value being half of the map (as shown in the following):

```python
self.base_top_left = (townhall.x < 32)
```

I'm thinking of removing this variable, since I don't know what would be the benefit of keeping the townhall in the top left corner of the grid. Is that okay?

Reply:
I think we can remove this variable, as it restricts you to one map. We should leave this class as generic as possible, because other classes may arise from it that may need to know the position of townhall.